sycl/include/sycl/ext/oneapi/weak_object.hpp

```diff
@@ -141,6 +151,78 @@ class weak_object<buffer<T, Dimensions, AllocatorT>>
   bool MIsSubBuffer;
 };
 
+// Specialization of weak_object for stream as it needs additional members
```

Comment:
This looks good, but I think we should leave a comment in buffer.hpp and stream.hpp that changing the class member vars might require a concomitant change of its specialization in this extension.  Not a likely occurrence, given both `buffer_plain` and `buffer_impl`, but still useful for anyone who may come along to make changes later. sycl::stream, particularly, seems vulnerable to that.

Reply:
I've added comments around their members stating that anything needed for reconstruction should also be in their weak_object specializations.